clients/gateway.go

```diff
@@ -8,25 +8,57 @@ import (
 	"net/http"
 	"net/url"
 	"strconv"
+	"time"
 
 	"github.com/NethermindEth/juno/core/felt"
 )
 
-const feederGatewayPath = "/feeder_gateway/"
+type Backoff func(wait time.Duration) time.Duration
 
 type GatewayClient struct {
-	baseUrl string
+	url        string
+	backoff    Backoff
+	maxRetries int
+	minWait    time.Duration
 }
 
-func NewGatewayClient(baseUrl string) *GatewayClient {
+func (c *GatewayClient) WithBackoff(b Backoff) *GatewayClient {
+	c.backoff = b
+	return c
+}
+
+func (c *GatewayClient) WithMaxRetries(d int) *GatewayClient {
+	c.maxRetries = d
```

Comment:
Trivial. But can we make this descriptive? Say, `num` as in number of retries?

Also maybe `uint` since we can't have negative retries
```suggestion
func (c *GatewayClient) WithMaxRetries(num uint) *GatewayClient {
	c.maxRetries = num
```